gammapy/analysis/config.py

```diff
@@ -66,127 +79,210 @@ class Config:
             Time: lambda v: f"{v.value}",
         }
 
-    @classmethod
-    def from_yaml(cls, filename):
-        config = read_yaml(filename)
-        return AnalysisConfig(**config)
-
-    def to_yaml(self):
-        return yaml.dump(yaml.safe_load(self.json()))
-
-    def update_from_dict(self, other):
-        data = deep_update(
-            self.dict(exclude_defaults=True), other.dict(exclude_defaults=True)
-        )
-        return AnalysisConfig(**data)
 
-
-class SkyCoordType(GammapyBaseModel):
-    frame: FrameEnum = FrameEnum.icrs
+class SkyCoordConfig(GammapyBaseConfig):
+    frame: FrameEnum = None
     lon: AngleType = None
     lat: AngleType = None
 
 
-class EnergyAxis(GammapyBaseModel):
+class EnergyAxisConfig(GammapyBaseConfig):
     min: EnergyType = "0.1 TeV"
     max: EnergyType = "10 TeV"
     nbins: int = 30
 
 
-class SpatialCircleRange(GammapyBaseModel):
-    frame: FrameEnum = FrameEnum.icrs
+class SpatialCircleRangeConfig(GammapyBaseConfig):
```

Comment:
Suggest to rename to `SpatialCircleConfig`, IMO a circle isn't a "range".

Reply:
Done